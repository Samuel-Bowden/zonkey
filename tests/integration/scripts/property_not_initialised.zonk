class Section {
	Boolean built;
	[Text] children;

	constructor() {}

	method build() -> [Text] {	
		for (let i = 0, i < @children.len(), i += 1) {
			@children.get(i).set_text(integer_to_string(i));
		}
		return @children;
	}
}

start {
	Section().build();
}
