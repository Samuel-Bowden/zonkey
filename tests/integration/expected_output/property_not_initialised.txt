(EXCEPTION) PropertyNotInitialised
  Attempted to access property with name 'children' without initialising it first.
	3 | 	[Text] <children>;
